ft232/mpsse.py

```diff
@@ -0,0 +1,555 @@
+
+    def clockWithoutDataUntilLow(self):
+        """ Missing description.
+        This function implements chapter 6.8 Clk continuously and Wait On I/O Low.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        if self.__isModernDevice():
+            self.__Ft232.write(bytes([0x95]))
+
+    def __enableAdaptiveClock(self):
+        """ Missing description.
+        This function implements chapter 6.9 Turn On Adaptive clocking.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        if self.__isModernDevice():
+            self.__Ft232.write(bytes([0x96]))
+
+    def __disableAdaptiveClock(self):
+        """ Missing description.
+        This function implements chapter 6.10 Turn Off Adaptive clocking.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        if self.__isModernDevice():
+            self.__Ft232.write(bytes([0x97]))
+
+    def clockWithoutDataUntilHighOrBytes(self, NumberBytes):
+        """ Missing description.
+        This function implements chapter 6.11 Clock For n x 8 bits with no data transfer or Until GPIOL1 is High.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        if self.__isModernDevice():
+            NumberLowByte = NumberBytes & 0xff
+            NumberHighByte = (NumberBytes >> 8) & 0xff
+            self.__Ft232.write(bytes([0x9c, NumberLowByte, NumberHighByte]))
+
+    def clockWithoutDataUntilLowOrBytes(self, NumberBytes):
+        """ Missing description.
+        This function implements chapter 6.12 Clock For n x 8 bits with no data transfer or Until GPIOL1 is Low.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        if self.__isModernDevice():
+            NumberLowByte = NumberBytes & 0xff
+            NumberHighByte = (NumberBytes >> 8) & 0xff
+            self.__Ft232.write(bytes([0x9d, NumberLowByte, NumberHighByte]))
+
+    def setIoDriveTristate(self): #feel free to change the name
+        """ Missing description.
+        This function implements chapter 7.1 Set I/O to only drive on a ‘0’ and tristate on a ‘1’.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        if self.__isModernDevice():
+            self.__Ft232.write(bytes([0x9e]))
+
+    def _sync(self):
+        pass
+
+    def __setBitModeMPSSE(self):
+        """ This sets the Ftx232 to MPSSE Mode.  
+        """
+        self.__Ft232.setBitMode(mask=0, mode=self.__Ft232.BM_MPSSE)
+
+    def __isModernDevice(self):
+        """ This checks if the board is a modern device. 
+        
+        Returns
+        -------
+        bool
+        """
+        return self.__Ft232.MPSSE_CLOCK == 60e6
```

Comment:
This seems a bit ugly. Don't base an attribute of the device on the clock frequency, you have no idea how this might change in the future. Better to just have a boolean in your "board" file indicating whether a feature is present or, better yet, have the boards define the features that apply to them (though this might require some restructuring). 

Also "ModernDevice" is a terrible name for anything (even if FTDI uses it themselves). What happens when they add new features? Do we call it a "MoreModernThanModernDevice"?